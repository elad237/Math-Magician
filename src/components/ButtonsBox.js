import React from 'react';
import PropTypes from 'prop-types';

class ButtonsBox extends React.PureComponent {
  render() {
    const { handleClick } = this.props;
    return (
      <div className="buttons">
        <button onClick={handleClick} type="submit">AC</button>
        <button onClick={handleClick} type="submit">+/-</button>
        <button onClick={handleClick} type="submit">%</button>
        <button onClick={handleClick} type="submit" className="operator">รท</button>
        <button onClick={handleClick} type="submit">7</button>
        <button onClick={handleClick} type="submit">8</button>
        <button onClick={handleClick} type="submit">9</button>
        <button onClick={handleClick} type="submit" className="operator">x</button>
        <button onClick={handleClick} type="submit">4</button>
        <button onClick={handleClick} type="submit">5</button>
        <button onClick={handleClick} type="submit">6</button>
        <button onClick={handleClick} type="submit" className="operator">-</button>
        <button onClick={handleClick} type="submit">1</button>
        <button onClick={handleClick} type="submit">2</button>
        <button onClick={handleClick} type="submit">3</button>
        <button onClick={handleClick} type="submit" className="operator">+</button>
        <button onClick={handleClick} type="submit" className="span-two">0</button>
        <button onClick={handleClick} type="submit">.</button>
        <button onClick={handleClick} type="submit" className="operator">=</button>
      </div>
    );
  }
}

ButtonsBox.propTypes = { handleClick: PropTypes.func.isRequired };

export default ButtonsBox;
