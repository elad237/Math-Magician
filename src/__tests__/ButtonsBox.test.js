import React from 'react';
import renderer from 'react-test-renderer';
import ButtonsBox from '../components/Home';

describe('Jest Snapshot testing suite', () => {
  it('Matches DOM Snapshot', () => {
    const tree = renderer.create(<ButtonsBox />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Matches DOM Snapshot', () => {
    const tree = renderer.create(<ButtonsBox handleClick="handleClick" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
