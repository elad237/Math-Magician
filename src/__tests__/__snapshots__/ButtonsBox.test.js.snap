// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jest Snapshot testing suite Matches DOM Snapshot 1`] = `
<div
  className="home-container"
>
  <h2
    className="home-title"
  >
    Welcome to our page!
  </h2>
  <p
    className="home-text"
  >
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean eu feugiat ex, et fermentum ipsum. Vestibulum ultrices dapibus enim egestas dictum. Suspendisse facilisis, purus et finibus tincidunt, nulla nulla eleifend mi, sit amet ullamcorper tellus ipsum in mi. Fusce imperdiet, augue nec iaculis congue, est mi accumsan elit, eget hendrerit risus leo a mauris. Donec id pretium eros, nec gravida metus. Vestibulum commodo turpis vel porta efficitur. Sed commodo, metus ut interdum euismod, nulla massa tempor mi, et suscipit mi quam vitae eros. Morbi nec lorem eget tortor tincidunt convallis. In tristique eleifend mauris ac maximus. Sed ut ultricies sapien, vitae fringilla eros.
    <br />
    <br />
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean eu feugiat ex, et fermentum ipsum. Vestibulum ultrices dapibus enim egestas dictum. Suspendisse facilisis, purus et finibus tincidunt, nulla nulla eleifend mi, sit amet ullamcorper tellus ipsum in mi. Fusce imperdiet, augue nec iaculis congue, est mi accumsan elit, eget hendrerit risus leo a mauris. Donec id pretium eros, nec gravida metus.
  </p>
</div>
`;

exports[`Jest Snapshot testing suite Matches DOM Snapshot 2`] = `
<div
  className="home-container"
>
  <h2
    className="home-title"
  >
    Welcome to our page!
  </h2>
  <p
    className="home-text"
  >
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean eu feugiat ex, et fermentum ipsum. Vestibulum ultrices dapibus enim egestas dictum. Suspendisse facilisis, purus et finibus tincidunt, nulla nulla eleifend mi, sit amet ullamcorper tellus ipsum in mi. Fusce imperdiet, augue nec iaculis congue, est mi accumsan elit, eget hendrerit risus leo a mauris. Donec id pretium eros, nec gravida metus. Vestibulum commodo turpis vel porta efficitur. Sed commodo, metus ut interdum euismod, nulla massa tempor mi, et suscipit mi quam vitae eros. Morbi nec lorem eget tortor tincidunt convallis. In tristique eleifend mauris ac maximus. Sed ut ultricies sapien, vitae fringilla eros.
    <br />
    <br />
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean eu feugiat ex, et fermentum ipsum. Vestibulum ultrices dapibus enim egestas dictum. Suspendisse facilisis, purus et finibus tincidunt, nulla nulla eleifend mi, sit amet ullamcorper tellus ipsum in mi. Fusce imperdiet, augue nec iaculis congue, est mi accumsan elit, eget hendrerit risus leo a mauris. Donec id pretium eros, nec gravida metus.
  </p>
</div>
`;
